base/daemon/BaseDaemon.cpp

```diff
@@ -81,13 +81,17 @@ static void call_default_signal_handler(int sig)
 
 static constexpr size_t max_query_id_size = 127;
 
+// equal to setting 'log_queries_cut_to_length'
+static constexpr size_t max_query_size = 100000;
```

Comment:
This is not correct, since you cannot write such many bytes to pipe atomically.

Reply:
See https://github.com/ClickHouse/ClickHouse/blob/2966e9aac4c39dbee6cd1aa3cde3607eaec1b45f/src/Common/TraceSender.cpp#L37-L40 for the reference

Reply:
End even though there is no SA_NODEFER, this handler is used for multiple signals, so if multiple of them will be arrived then the information in pipe can be overlapped, yes this is not that likely, but does it worth it?

Reply:
> End even though there is no SA_NODEFER, this handler is used for multiple signals, so if multiple of them will be arrived then the information in pipe can be overlapped, yes this is not that likely, but does it worth it?

Thanks, maybe it is just only for special use.